src/gentropy/method/l2g/trainer.py

```diff
@@ -155,148 +177,290 @@ def log_to_wandb(
+            self.x_train is None
+            or self.x_test is None
+            or self.y_train is None
+            or self.y_test is None
+            or self.features_list is None
         ):
-            assert (
-                not self.x_train.empty and not self.y_train.empty
-            ), "Train data not set, nothing to evaluate."
-            fitted_classifier = self.model.model
-            y_predicted = fitted_classifier.predict(self.x_test.values)
-            y_probas = fitted_classifier.predict_proba(self.x_test.values)
-            self.run = wandb_init(
-                project=self.wandb_l2g_project_name,
-                name=wandb_run_name,
-                config=fitted_classifier.get_params(),
-            )
-            # Track classification plots
-            plot_classifier(
-                self.model.model,
-                self.x_train.values,
-                self.x_test.values,
-                self.y_train,
-                self.y_test,
-                y_predicted,
-                y_probas,
-                labels=list(self.model.label_encoder.values()),
-                model_name="L2G-classifier",
-                feature_names=self.features_list,
-                is_binary=True,
-            )
-            # Track evaluation metrics
-            self.run.log(
-                {
-                    "areaUnderROC": roc_auc_score(
-                        self.y_test, y_probas[:, 1], average="weighted"
-                    )
-                }
-            )
-            self.run.log({"accuracy": accuracy_score(self.y_test, y_predicted)})
-            self.run.log(
-                {
-                    "weightedPrecision": precision_score(
-                        self.y_test, y_predicted, average="weighted"
-                    )
-                }
-            )
-            self.run.log(
-                {
-                    "weightedRecall": recall_score(
-                        self.y_test, y_predicted, average="weighted"
-                    )
-                }
-            )
-            self.run.log({"f1": f1_score(self.y_test, y_predicted, average="weighted")})
-            # Track gold standards and their features
-            self.run.log(
-                {"featureMatrix": Table(dataframe=self.feature_matrix._df.toPandas())}
-            )
-            # Log feature missingness
-            self.run.log(
-                {
-                    "missingnessRates": self.feature_matrix.calculate_feature_missingness_rate()
-                }
-            )
-            # Plot marginal contribution of each feature
-            explanation = self._get_shap_explanation(self.model)
-            self.log_plot_image_to_wandb(
-                "Feature Contribution",
-                shap.plots.bar(
-                    explanation, max_display=len(self.x_train.columns), show=False
-                ),
-            )
+            raise ValueError("Something went wrong, couldn't log to W&B.")
```

Comment:
The error message does not tell what went wrong, a bit misleading.

Reply:
resolved in https://github.com/opentargets/gentropy/pull/938/commits/b3e36ba6d310c52ab99a4441fadd7f30489d0bc9